spec/models/recording_spec.rb

```diff
@@ -15,6 +15,22 @@
     it { is_expected.to validate_presence_of(:participants) }
   end
 
+  describe 'scopes' do
+    context 'with_provider' do
+      it 'only includes users with the specified provider' do
+        room1 = create(:room, user: create(:user, provider: 'greenlight'))
+        room2 = create(:room, user: create(:user, provider: 'test'))
+
+        create_list(:recording, 5, room: room1)
+        create_list(:recording, 5, room: room2)
+
+        recs = described_class.includes(:user).with_provider('greenlight')
+        expect(recs.count).to eq(5)
```

Comment:
I think these can be merged into one if we do something like `expect(recs.pluck(:id)).to match_array(greenlight_recs.pluck(:id))`

Reply:
I prefer doing an explicit check to confirm that all providers are what we expect

Reply:
(in case something happens in the factory for some reason)